src/ncp-spinel/SpinelNCPInstance.cpp

```diff
@@ -959,6 +983,24 @@ SpinelNCPInstance::set_property(
 				.finish()
 			);
 
+		} else if (strcaseequal(key.c_str(), kWPANTUNDProperty_BorderAgentProxyStream)) {
+			Data packet = any_to_data(value);
+
+			uint16_t port = *(uint16_t*)(packet.data() + packet.size() - sizeof(port));
```

Comment:
The pointer type-cast model used here is not really safe to memory alignments. 
Better to read each bytes from packet/array and then convert the values to the type.  We also need to be careful of endian-ness of host (e.g., how `uint16` is read from memory may be different on different platforms)...